psydac/api/feec.py

```diff
@@ -430,124 +429,166 @@ def projectors(self, *, kind='global', nquads=None):
+
+                self._conf_proj = (cP0, cP1, cP2)
 
         elif self.dim == 3:
             raise NotImplementedError("3D projectors are not available")
 
-        return cP
+        if kind == 'femlinop':
+            return cP0, cP1, FemLinearOperator(fem_domain=self.V2, fem_codomain=self.V2, linop=cP2, sparse_matrix=cP2.tosparse)
+        elif kind == 'sparse':
+            return cP0.tosparse, cP1.tosparse, cP2.tosparse
+        elif kind == 'linop': 
+            return cP0.linop, cP1.linop, cP2
 
     #--------------------------------------------------------------------------
-    def hodge_operator(self, space=None, backend_language='python', load_dir=None):
+    def init_Hodge_operator(self, backend_language='python', load_dir=None):
         """
-        return the conforming projectors of the broken multi-patch space
+        Initialize the Hodge operator for the multipatch de Rham sequence.
 
         Parameters
         ----------
-        space : <str>
-          The space of the projector
 
         backend_language: <str>
           The backend used to accelerate the code
 
         load_dir: <str|None>
           Filename for storage in sparse matrix format
 
-        Returns
-        -------
-        H: <FemLinearOperator>
-          The Hodge operator
-
         """
 
-        H = None
         if self.dim == 1:
             raise NotImplementedError("1D Hodges are not available")
 
         elif self.dim == 2:
-            if space == 'V0':
-                H = HodgeOperator(self.V0, self.domain_h, backend_language=backend_language, load_dir=load_dir, load_space_index=0)
-            elif space == 'V1':
-                H = HodgeOperator(self.V1, self.domain_h, backend_language=backend_language, load_dir=load_dir, load_space_index=1)
-            elif space == 'V2':
-                H = HodgeOperator(self.V2, self.domain_h, backend_language=backend_language, load_dir=load_dir, load_space_index=2)
-
-            elif space == None and self.sequence[1] == 'hcurl': 
+            if not self._Hodge_operators: 
+
                 H0 = HodgeOperator(self.V0, self.domain_h, backend_language=backend_language, load_dir=load_dir, load_space_index=0)
                 H1 = HodgeOperator(self.V1, self.domain_h, backend_language=backend_language, load_dir=load_dir, load_space_index=1)
                 H2 = HodgeOperator(self.V2, self.domain_h, backend_language=backend_language, load_dir=load_dir, load_space_index=2)
 
-                return H0, H1, H2
-            else:
-                raise ValueError('Invalid value for "space" argument: {}'.format(space))
+                self._Hodge_operators = (H0, H1, H2)
 
         elif self.dim == 3:
             raise NotImplementedError("3D Hodgesa are not available")
 
-        return H
+    #--------------------------------------------------------------------------
+    def Hodge_kind(self, H, dual=False, kind='femlinop'):
```

Comment:
I am not convinced by adding `_kind` in this function's name -- unless there's a `Hodge_nasty` somewhere else

Reply:
Do you have other naming suggestions? This was the first one that came to my mind.